pkg/cli/login/loginoptions.go

```diff
@@ -262,6 +280,52 @@ func (o *LoginOptions) gatherAuthInfo() error {
 		}
 	}
 
+	if o.OIDCExecPluginType == string(OC) {
+		issuerUrl, err := o.extractIssuerURLForOIDC()
+		if err != nil {
+			return err
+		}
+
+		clientConfig.ExecProvider = &kclientcmdapi.ExecConfig{
+			APIVersion: clientauthentication.GroupName + "/v1",
+			Command:    "oc",
+			Args: []string{
+				"get-token",
+				fmt.Sprintf("--issuer-url=%s", issuerUrl),
+				fmt.Sprintf("--client-id=%s", o.OIDCClientID),
+				fmt.Sprintf("--callback-addresses=localhost:%d", o.CallbackPort),
+			},
+			InstallHint:     "Please be sure that oc is defined in $PATH to be executed as credentials exec plugin",
+			InteractiveMode: kclientcmdapi.IfAvailableExecInteractiveMode,
+		}
+
+		if len(o.OIDCExtraScopes) > 0 {
+			clientConfig.ExecProvider.Args = append(clientConfig.ExecProvider.Args, fmt.Sprintf("--extra-scopes=%s", strings.Join(o.OIDCExtraScopes, ",")))
+		}
+
+		if o.OIDCClientSecret != "" {
+			clientConfig.ExecProvider.Args = append(clientConfig.ExecProvider.Args, fmt.Sprintf("--client-secret=%s", o.OIDCClientSecret))
+		}
+
+		if o.InsecureTLS {
+			clientConfig.ExecProvider.Args = append(clientConfig.ExecProvider.Args, "--insecure-skip-tls-verify")
+		}
+
+		if len(o.Config.CAFile) > 0 {
+			clientConfig.ExecProvider.Args = append(clientConfig.ExecProvider.Args, fmt.Sprintf("--certificate-authority=%s", o.Config.CAFile))
+		}
+
+		me, err := project.WhoAmI(clientConfig)
+		if err != nil {
+			return err
+		}
+
+		o.Username = me.Name
+		o.Config = clientConfig
+		fmt.Fprintf(o.Out, "Logged into %q as %q from the external oidc issuer.\n\n", o.Config.Host, o.Username)
+		return nil
```

Comment:
extract to a separate function?

Reply:
ExecProvider setup process is extracted to separate function.